week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -39,12 +39,39 @@ DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
+var destinationNamesWithin500Kms = travelDestinations.filter(getNamesin500).map(getName);
 
-var destinationNamesWithin500Kms = // Complete here
+function getNamesin500(destination){
+    return (destination.distanceKms<500);
+}
 
-var destinationNameReachableByFerry = // Complete here
+function getName(destination){
+    return destination.destinationName;
+}    
 
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+// Complete here
+
+var destinationNameReachableByFerry = travelDestinations.filter(getNamebyFerry).map(getName);
+
+function getNamebyFerry(destination){
+    return (destination.transportations.includes("ferry"));
+}
+
+// Complete here
+
```

Comment:
It would be much better to assemble this `destinationNamesMoreThan300KmsAwayByTrain` list using only `filter` and `map`.  There is no need to use `forEach` until you have to print the final results (you've missed that bit!)